core/controllers/question_editor.py

```diff
@@ -38,19 +38,20 @@ def post(self):
         if (
                 ('id' in question_dict) or
                 ('question_state_data' not in question_dict) or
-                ('language_code' not in question_dict) or
-                ('question_state_schema_version' not in question_dict)):
+                ('language_code' not in question_dict)):
```

Comment:
This change seems wrong. The schema version should be passed in from the frontend, and the backend should verify that it is the correct one (like it used to before).

It is possible for the backend schema version to update while an editing session is in progress, and we need to catch these errors rather than sticking a possibly-incorrect schema version onto the passed-in question dict.

Reply:
Yeah -- we really should, though. Currently exploration drafts are actually broken after an exploration migration, because they do not take versioning into account.

You could do an async call to retrieve the schema version. Your method also sounds fine to me, but we should have just one constant in the entire codebase. I think I lean a bit towards the async call since it doesn't expose the schema version in other places needlessly.

Reply:
Done, added an async backend call to fetch schema version and renamed EXPLORATION_STATES_SCHEMA_VERSION to STATES_SCHEMA_VERSION.